clap_derive/src/lib.rs

```diff
@@ -15,6 +15,7 @@
 #![doc(html_logo_url = "https://raw.githubusercontent.com/clap-rs/clap/master/assets/clap.png")]
 #![doc = include_str!("../README.md")]
 #![forbid(unsafe_code)]
+#![deny(clippy::all)]
```

Comment:
1. We intentionally don't `deny` in the files so people can see all failures and iterate without being blocked.  `deny` causes a fail-fast.
2. What is the intention with `all`?  By going for all, local builds can randomly brake when someone upgrades before we update the code base, making it harder to contribute

Reply:
I think in practice the number of breaking changes introduced this way is low, and not difficult to manage.

On the other hand, having this compiler flag makes the behaviour consistent between CI and local builds.

If you prefer to sit on the other side of that trade-off I can remove this

Reply:
For people who want to reproduce CI locally, we provide a [Makefile](https://github.com/clap-rs/clap/blob/master/Makefile) so they can easily reproduce our specific feature flag combinations we test.  It also helps with this.